mem0/memory/main.py

```diff
@@ -113,7 +115,7 @@ def add(
         user_id=None,
         agent_id=None,
         run_id=None,
-        metadata=None,
+        metadata: dict | None = None,
```

Comment:
Change this to `Optional[dict]` because this project supports Python >=3.9 and `|` was introduced in Python 3.10